test/e2e/custom_tool_test.go

```diff
@@ -350,3 +350,60 @@ func TestPreserveFileModeForCMP(t *testing.T) {
 			assert.Equal(t, "ConfigMap", app.Status.Resources[0].Kind)
 		})
 }
+
+func TestCMPWithSymlinkPartialFiles(t *testing.T) {
+	Given(t).
+		And(func() {
+			go startCMPServer("./testdata/cmp-symlink")
+			time.Sleep(1 * time.Second)
+			os.Setenv("ARGOCD_BINARY_NAME", "argocd")
+		}).
+		Path("guestbook-partial-symlink-files").
+		When().
+		CreateFromFile(func(app *Application) {
+			app.Spec.Source.Plugin = &ApplicationSourcePlugin{Name: "cmp-symlink-v1.0"}
```

Comment:
I plugin name is provided then discovery is not used: repo server connects to the specified CMP plugin server. We need to test without plugin name to confirm discovery is fixed.

Reply:
I have updated test with the discovery setting, please take a look